app/helpers/application_helper.rb

```diff
@@ -31,8 +31,9 @@ def ial2_requested?
   end
 
   def liveness_checking_enabled?
-    FeatureManagement.liveness_checking_enabled? &&
-      (sp_session[:issuer].blank? || sp_session[:ial2_strict])
+    return false if !FeatureManagement.liveness_checking_enabled?
+    return sp_session[:ial2_strict] if sp_session.key?(:ial2_strict)
+    !!current_user && current_user.profiles.verified.any?(&:includes_liveness_check?)
```

Comment:
do we want any previous or just the most recently active?

Reply:
> do we want any previous or just the most recently active?

I'm not entirely sure since I don't understand the real-world use-cases of multiple profiles. I guess if the result of the reverification is that the user's most recent profile would be restored, then it should match the most recent? I'll check to see what happens at the end of the flow to make sure I'm understanding this right.

Reply:
> > do we want any previous or just the most recently active?
> 
> I'm not entirely sure since I don't understand the real-world use-cases of multiple profiles. I guess if the result of the reverification is that the user's most recent profile would be restored, then it should match the most recent? I'll check to see what happens at the end of the flow to make sure I'm understanding this right.

I dug into this a bit more:

- Reverification after a password recovery doesn't "restore" a profile, it creates a new one.
- If the user has their personal key, the profile restored is the most recent activated one ([`UserDecorator#password_reset_profile`](https://github.com/18F/identity-idp/blob/41583e6ae094257ddfe54bf2e5433b17546d0462/app/decorators/user_decorator.rb#L111-L116)).

So if we expect the only way to get to this point is via password recovery, I think it's reasonable to expect the default (or at least enforcement of) would be based on this same logic. Will confirm with product just to be sure, but otherwise changes implemented in e7d05d1.

Reply:
> Will confirm with product just to be sure, but otherwise changes implemented in [e7d05d1](https://github.com/18F/identity-idp/commit/e7d05d1631665ae16165068299a502007d84c36b) .

https://gsa-tts.slack.com/archives/CNCGEHG1G/p1626460205007300